subprojects/plugins/src/main/java/org/gradle/api/internal/java/JavaLibrary.java

```diff
@@ -112,87 +106,21 @@ private UsageContext createCompileUsageContext() {
     }
 
     @Override
-    public void addFeatureVariantsFromConfiguration(Configuration outgoingConfiguration, Spec<? super ConfigurationVariant> spec) {
+    public void addFeatureVariantsFromConfiguration(Configuration outgoingConfiguration, final Spec<? super ConfigurationVariant> spec) {
```

Comment:
⭕️ Given the changes in this PR, the recently introduced `ComponentWithFeatures` can be removed completely.

Reply:
Removed `ComponentWithFeatures`, good catch!